app/controllers/order/client_information.js

```diff
@@ -57,34 +63,87 @@ export default Ember.Controller.extend(cancelOrder, {
     return this.get('selectedId') === 'hkId' ? 1 : 2;
   },
 
+  actionType(isOrganisation, beneficiaryId) {
+    let actionType;
+    if (isOrganisation && beneficiaryId) {
+      actionType = 'DELETE';
+    } else if (!isOrganisation && beneficiaryId) {
+      actionType = 'PUT';
+    } else if (!isOrganisation && !beneficiaryId) {
+      actionType = 'POST';
+    } else {
+      actionType = null;
+    }
+    return actionType;
+  },
+
   actions: {
-    saveClientDetails(){
-      var orderId = this.get('order.id');
-      var beneficiaryId = this.get('beneficiary.id');
+    saveClientDetails() {
+      let orderParams;
+      let clientInfo = this.get('clientInfoId');
 
-      var url, actionType;
+      const isForOrganisation = clientInfo === 'organisation';
+      orderParams = isForOrganisation ? {
+          'purpose_ids': [1],
```

Comment:
Can we load purposes in ember-data and then use it here? Let me know your thoughts.

Reply:
Yes.. that will be more reliable. Will add that in ember-data post-login.